### PR TITLE
Using latest iid library from firebase

### PR DESCRIPTION
**Pull request by andresesfm**
## Description
iid library: com.google.android.gms:play-services-iid has been deprecated as per: https://developers.google.com/android/reference/com/google/android/gms/iid/package-summary
It has been replaced with firebase
Fixed issue #<issue-number>

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅     |
| Windows |    ❌     |

## Checklist

* [X ] I have tested this on a device/simulator for each compatible OS
* [ ] I added the documentation in `README.md`
* [ ] I mentioned this change in `CHANGELOG.md`
* [ ] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [ ] I updated the dummy web/test polyfill (`default/index.js`)
* [ ] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1406,11 +1406,11 @@ Fired when the battery level changes; sent no more frequently than once per minu
 #### Examples
 
 ```js
-import { useBatteryLevel } from 'react-native-device-info'
+import { useBatteryLevel } from 'react-native-device-info';
 
 const batteryLevel = useBatteryLevel(); // 0.759999
 
-<Text>{ batteryLevel }</Text>
+<Text>{batteryLevel}</Text>;
```
Comment (mikehardy) [inline README.md]: I like the semi-colon additions but you've altered whitespace at the same time, always a no-no for drive-by contributions ;-)

File: android/build.gradle [hunk 2]
```diff
@@ -45,6 +45,6 @@ repositories {
 
 dependencies {
     implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
-    def iidVersion = safeExtGet('googlePlayServicesIidVersion', safeExtGet('googlePlayServicesVersion', '17.0.0'))
-    implementation "com.google.android.gms:play-services-iid:$iidVersion"
+    def iidVersion = safeExtGet('iidVersion', safeExtGet('iidVersion', '19.0.1'))
```
Comment (mikehardy) [inline android/build.gradle]: That's a breaking change, not worth it in my opinion

File: android/build.gradle [hunk 3]
```diff
@@ -45,6 +45,6 @@ repositories {
...
+    implementation "com.google.firebase:firebase-iid:$iidVersion"
```
Comment (mikehardy) [inline android/build.gradle]: did IID move completely to firebase? I'm using firebase heavily but hadn't noticed! Would explain the change above, but to make it non-breaking you have to chain the safeExtGet's so that it falls back to the existing one and doesn't break the installed userbase

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 4]
```diff
@@ -724,11 +724,11 @@ public double getTotalMemorySync() {
   @SuppressWarnings({"ConstantConditions", "deprecation"})
   public String getInstanceIdSync() {
     try {
-      if (Class.forName("com.google.android.gms.iid.InstanceID") != null) {
-        return com.google.android.gms.iid.InstanceID.getInstance(getReactApplicationContext()).getId();
+      if (Class.forName("com.google.firebase.iid.FirebaseInstanceId") != null) {
+        return com.google.firebase.iid.FirebaseInstanceId.getInstance().getId();
```
Comment (mikehardy) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: I'll have to do some research quickly to make sure these are exactly equivalent - but in your research is that your conclusion? People rely on these ids being exact + precise for device id

**Comment by mikehardy** [comment]
This looks really clean now and I verified that yes indeed - we're all moving IIDs to firebase now, thanks for this!

**Comment by miblanchard** [comment]
@mikehardy Any idea when a new version with these changes will make it to npm?

**Comment by mikehardy** [comment]
I just did a huge batch today and I'm out of time, it might be a day or three sorry - there is always patch-package in the meantime - or I suppose since it's merged and committed you can depend on github using the commit hash then as a postinstall step run the build inside the package in node_modules to go from typescript -> javascript and have it be ready for use

**Comment by miblanchard** [comment]
That timing works for us. Thank for all your hard work in the community!
